web-quiz-app/src/components/NumericalQuestion.jsx

```diff
@@ -1,20 +1,113 @@
-import './css/NumericalQuestion.css'
+import './css/NumericalQuestion.css';
+import QuizDataService from "../services/quiz.service";
+import React, { useEffect, useState } from 'react';
+
+function NumericalQuestion(props) {
+    const [totalMarks, setMarks] = useState();
+    const [userID, setUserID] = useState();
+    const [quizID, setQuizID] = useState();
+    const [question, setQuestion] = useState('');
+    const [solution, setSolution] = useState('');
+    const [questionID, setQuestionID] = useState();
+    const [loadQuestion, setLoad] = useState(false);
+
+    useEffect(() => {
+        setUserID(props.insID)
+        setQuizID(props.qID)
+        if(props.load) {
+            setQuestionID(props.questionID)
+            setQuestion(props.question)
+            setSolution(props.solution)
+            setMarks(props.marks)
+            setLoad(true)
+        }
+    }, [])
+
+    useEffect(() => {
+        if(totalMarks != null) {
+            if(totalMarks < 0 || isNaN(totalMarks)) {
+                document.getElementById("totalMark").value = 0
+                setMarks(0)
+                alert("Total marks must be a positive integer")
+            }
+        }
+    }, [totalMarks])
+
+    function setTotalMarks(input) {
+        setMarks(input)
+    }
+
+    function setQuestionName(input) {
+        setQuestion(input)
+    }
+
+    function setSolutionResult(input) {
+        setSolution(input)
+    }
+
+    function submit() {
+        createQuestion()
+    }
+
+    function createQuestion() {
+        console.log(quizID)
+        var questionToCreate = {
+            question_data: 4,
```

Comment:
```suggestion
            question_data: 1,
```
The question data for numerical questions should be 1.